.circleci/config.yml

```diff
@@ -98,9 +98,49 @@ jobs:
           command: bundle exec danger || echo "danger failed"
           when: always # Run this even when tests fail
 
+  "Docs":
+    docker:
+      - image: circleci/ruby:2.3
+    steps:
+      - checkout
+      
+      - restore_cache:
+          key: v1-ubuntu-gems-{{ checksum "Gemfile" }}
+      - run:
+          name: Setup Build
+          command: |
+            bundle check --path .bundle || bundle install --jobs=4 --retry=3 --path .bundle
+      - save_cache:
+          key: v1-ubuntu-gems-{{ checksum "Gemfile" }}
+          paths:
+            - .bundle
+
+      - run: bundle exec fastlane validate_docs
+
+  "Source Code Linting":
+    docker:
+      - image: circleci/ruby:2.3
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: v1-ubuntu-gems-{{ checksum "Gemfile" }}
+      - run:
+          name: Setup Build
+          command: |
+            bundle check --path .bundle || bundle install --jobs=4 --retry=3 --path .bundle
+      - save_cache:
+          key: v1-ubuntu-gems-{{ checksum "Gemfile" }}
+          paths:
+            - .bundle
+
+      - run: bundle exec fastlane verify_source
+
 workflows:
   version: 2
   build:
     jobs:
-      - "macOS"
-      - "Ubuntu"
+      - "Tests on macOS"
```

Comment:
Can naming be more consistent?
1. when making those private lanes public we have a chance to rename them before they become a public API.
2. can the job names and their corresponding lanes be more consistent?

a proposal:
* Test on Ubuntu -> Run Tests on Ubuntu and lane becomes `run_tests`
* Docs -> Validate docs
* Source Code Linting -> Lint Source or Verify Source (and make lane named appropriately)

Reply:
> https://docs.fastlane.tools/actions/run_tests/

Reply:
> 1. when making those private lanes public we have a chance to rename them before they become a public API.
> 2. can the job names and their corresponding lanes be more consistent?

Sure, both possible. But using same names for the CI "containers" and fastlane lanes could imply a connection that is not 100% there. There is happening more then just executing the the `execute_tests` lane.

> - Test on Ubuntu -> Run Tests on Ubuntu and lane becomes run_tests

Unfortunately not: https://docs.fastlane.tools/actions/run_tests/

> Docs -> Validate docs

What does `Validate` really mean? How is it different from `Verify`?

For the Container names I aimed for "first word shows the important thing" with "Test, Docs, Source Code".

Reply:
@janpio I think what @lacostej might be getting at is that the naming is somewhat inconsistent:

- Docs: noun
- Source Code Linting: noun + verb in -ing form
- Tests on Ubuntu/macOS: noun + on + noun

How about something like this:

- Validating Docs 
- Linting Source
- Testing Ubuntu/macOS
- (Testing ruby 2.0/2.1)

Reply:
and the lane names then?

Reply:
Lane names are fine I think, since this is just for the Fastfile used in fastlane anyway. No one is going to copy this